Core/Services/AuthenticationService.cs

```diff
@@ -88,42 +88,59 @@ public async Task<ServiceResponse> Register(RegisterRequest model)
                 return new ErrorResponse { StatusCode = HttpStatusCode.UnprocessableEntity, Message = CreateValidationErrorMessage(result) };
 
             await _userManager.AddToRoleAsync(user, UserRoles.User);
-            var userFromDb = await _userManager.FindByNameAsync(user.UserName);
 
+            var userFromDb = await _userManager.FindByNameAsync(user.UserName);
             var token = await _userManager.GenerateEmailConfirmationTokenAsync(userFromDb);
+            var urlString = BuildUrl(token, userFromDb.UserName, _config["ReturnPaths:ConfirmEmail"]);
 
-            var uriBuilder = new UriBuilder(_config["ReturnPaths:ConfirmEmail"]);
-            var query = HttpUtility.ParseQueryString(uriBuilder.Query);
-            query["token"] = token;
-            query["username"] = userFromDb.UserName;
-            uriBuilder.Query = query.ToString();
-            var urlString = uriBuilder.ToString();
-
-            var senderEmail = _config["ReturnPaths:SenderEmail"];
-            await _emailService.SendEmailAsync(senderEmail, userFromDb.Email, "Confirm your email address", urlString);
+            await _emailService.SendEmailAsync(_config["ReturnPaths:SenderEmail"], userFromDb.Email, "Confirm your email address", urlString);
 
             return new SuccessResponse { StatusCode = HttpStatusCode.Created, Message = "User created successfully! Confirm your email." };
         }
-        public async Task<ServiceResponse> ConfirmEmail(ConfirmEmailModel model)
+
+        public async Task<ServiceResponse> ConfirmEmail(EmailDto model)
         {
             var user = await _userManager.FindByNameAsync(model.UserName);
             var result = await _userManager.ConfirmEmailAsync(user, model.Token);
 
-            if (model.UserName == null || user  == null || user.EmailConfirmed || !result.Succeeded)
+            if (model.UserName == null || user  == null || !result.Succeeded)
```

Comment:
You can shorten it to only check for result.Succeeded, as it will be false when user is null